contracts/Gauge.sol

```diff
@@ -383,162 +448,242 @@ contract Gauge is BaseGauge, IGauge {
-        return true;
+    function deposit(uint256 _assets, address _receiver)
+        external
+        returns (uint256)
+    {
+        _deposit(_assets, _receiver);
+        return _assets;
     }
 
-    function _deposit(address _for, uint256 _amount)
-        internal
-        updateReward(_for)
+    /** @notice deposit vault tokens into the gauge for a user
+     *   @dev vault token is taken from msg.sender
+     *   @dev This call update  `_for` claimable rewards
+     *   @dev shares and
+     *   @param _shares to deposit
+     *   @param _receiver the account to deposit to
+     *   @return amount of shares transfered
+     */
+    function mint(uint256 _shares, address _receiver)
+        external
+        returns (uint256)
     {
-        require(_amount != 0, "RewardPool : Cannot deposit 0");
-        if (_for != msg.sender) {
-            require(approvedTo[msg.sender][_for].deposit, "not allowed");
-        }
+        _deposit(_shares, _receiver);
+        return _shares;
+    }
 
-        //also deposit to linked rewards
-        uint256 length = extraRewards.length;
-        for (uint256 i = 0; i < length; ++i) {
-            IExtraReward(extraRewards[i]).rewardCheckpoint(_for);
-        }
+    function _deposit(uint256 _assets, address _receiver) internal {
+        require(_assets != 0, "RewardPool : Cannot deposit 0");
 
-        //give to _for
-        Balance storage balance = _balances[_for];
-        balance.lastDeposit = block.number;
+        //take away from sender
+        asset.safeTransferFrom(msg.sender, address(this), _assets);
 
-        _totalSupply += _amount;
-        uint256 newBalance = balance.realBalance + _amount;
-        balance.realBalance = newBalance;
-        uint256 boostedBalance = _boostedBalanceOf(_for, newBalance);
-        balance.boostedBalance = boostedBalance;
-        emit BoostedBalanceUpdated(_for, boostedBalance);
+        // mint shares
+        _mint(_receiver, _assets);
 
-        //take away from sender
-        stakingToken.safeTransferFrom(msg.sender, address(this), _amount);
-        emit Staked(_for, _amount);
+        emit BoostedBalanceUpdated(_receiver, _boostedBalances[_receiver]);
+        emit Deposit(msg.sender, _receiver, _assets, _assets);
+    }
+
+    /**
+      Maximum amount of the underlying asset that can be withdrawn from the owner balance in the Vault, through a withdraw call.
+    */
+    function maxWithdraw(address _owner) external view returns (uint256) {
+        return balanceOf(_owner);
     }
 
-    /** @notice allow an address to deposit on your behalf
+    function previewWithdraw(uint256 _assets) external view returns (uint256) {
+        return _assets;
+    }
+
+    /** @notice allow an address to lock and claim on your behalf
+     * claim ermai
      *  @param _addr address to change approval for
-     *  @param _canDeposit can deposit
      *  @param _canClaim can deposit
      *  @return true
      */
     function setApprovals(
```

Comment:
i think it's better UX to have granular methods, instead of a single setter for boolean arguments where one argument can easily be sent in bad order and confusing for caller.

`setApprovals(0x01, true, false)` can have different consequences than `setApprovals(0x01, false, true)`

potentially error prone, i prefer explicit methods.

`enableClaimingFrom(address)`. `disableClaimingFrom(address)`

`enableLockingFrom(address)`. `disableLockingFrom(address)`

less arguments and makes the side effect explicit avoiding potential oops factor